microsoft/terminal
# Upgrade to check-spelling v0.0.24

## Summary of the Pull Request

This upgrades to [v0.0.24](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.24).

A number of GitHub APIs are being turned off shortly, so you need to upgrade or various uncertain outcomes will occur.

There are some minor bugs that I'm aware of and which I've fixed since this release (including a couple I discovered while preparing this PR).

## References and Relevant Issues

## Detailed Description of the Pull Request / Additional comments

There's a new accessibility forbidden pattern:

#### Should be `cannot` (or `can't`)

See https://www.grammarly.com/blog/cannot-or-can-not/
> Don't use `can not` when you mean `cannot`. The only time you're likely to see `can not` written as separate words is when the word `can` happens to precede some other phrase that happens to start with `not`.
> `Can't` is a contraction of `cannot`, and it's best suited for informal writing.
> In formal writing and where contractions are frowned upon, use `cannot`.
> It is possible to write `can not`, but you generally find it only as part of some other construction, such as `not only . . . but also.`
- if you encounter such a case, add a pattern for that case to patterns.txt.
```
\b[Cc]an not\b
```

It's not enabled because I'm waiting for:
- [ ] #18258 

## Validation Steps Performed

https://github.com/check-spelling-sandbox/terminal/actions/
https://github.com/check-spelling-sandbox/terminal/actions/runs/12076096237#summary-33676944561 is what you would have seen if this was running w/o the fixes included
https://github.com/check-spelling-sandbox/terminal/actions/runs/12032865816#summary-33545680702 is what I worked from
## PR Checklist
   - If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
